cloud/services/domains.go

```diff
@@ -62,6 +73,74 @@ func EnsureDNSEntries(ctx context.Context, mscope *scope.MachineScope, operation
 	return nil
 }
 
+// EnsureAkamaiDNSEntries ensures the domainrecord on Akamai EDGE DNS is created, updated, or deleted based on operation passed
+func EnsureAkamaiDNSEntries(ctx context.Context, mscope *scope.MachineScope, operation string, dnsEntries []DNSOptions) error {
+	// SetUp Client for Akamai EDGE DNS
+
+	fqdn := mscope.LinodeCluster.ObjectMeta.Name + "-" + mscope.LinodeCluster.Spec.Network.DNSUniqueIdentifier + "." + mscope.LinodeCluster.Spec.Network.DNSRootDomain
+
```

Comment:
I think it might be worth doing something like
`akaDNSClient := mscope.AkamaiDomainsClient`
and then just calling the Record functions on that without the extra functions below.

Reply:
Done